commands/publish/index.js

```diff
@@ -685,7 +689,9 @@ class PublishCommand extends Command {
     };
     const mapper = pkg => {
       const spec = `${pkg.name}@${pkg.version}`;
-      const distTag = getDistTag(pkg.get("publishConfig"));
+      const preDistTag = this.getPreDistTag(pkg);
+      const distTag = preDistTag || getDistTag(pkg.get("publishConfig"));
+      opts.tag = preDistTag || opts.tag;
```

Comment:
We really shouldn't be mutating the outer `opts.tag` inside the mapper. The only value that matters here is `distTag`, the parameter passed to `npmDistTag.add()`, `opts.tag` _should_ never be considered when the underlying code executes (it's always overwritten by the parameter [here](https://github.com/lerna/lerna/blob/4e3f8d8087e0e2e8e83c7ea79558c517203583c2/utils/npm-dist-tag/npm-dist-tag.js#L29)).

Reply:
Good catch, I wasn't understanding that the config snapshot should continue to show the `dist-tag` and ignore `pre-dist-tag` (as they can be used together).

Reply:
Fix: https://github.com/lerna/lerna/pull/1991/commits/850bc22e60d6fb7cac7c5543770a0cb3347bbf62